kernel/kselftest.py

```diff
@@ -80,10 +80,19 @@ def setUp(self):
         if self.run_type == 'upstream':
             location = self.params.get('location', default='https://github.c'
                                        'om/torvalds/linux/archive/master.zip')
-            tarball = self.fetch_asset("kselftest.zip", locations=[location],
-                                       expire='1d')
-            archive.extract(tarball, self.workdir)
-            self.buldir = os.path.join(self.workdir, 'linux-master')
+            path = ''
+            if location in ['.zip', 'tar']:
```

Comment:
`if "https://github.com/torvalds/linux/archive/master.zip" in [".zip", ".tar"]` won't work.

Reply:
ah right i was overconfident on this so no tested and it not just one line change, let me send commit for that